provider/pkg/provider/helm_release_test.go

```diff
@@ -0,0 +1,97 @@
+package provider
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func Test_MergeMaps(t *testing.T) {
+	m := map[string]interface{}{
+		"a": map[string]interface{}{
+			"b": map[string]interface{}{
+				"d": []interface{}{
+					"1", "2",
+				},
+			},
+		},
+	}
+
+	override := map[string]interface{}{
+		"a": map[string]interface{}{
+			"b": map[string]interface{}{
+				"d": []interface{}{
+					"3", "4",
+				},
+			},
+		},
+	}
+
+	for _, test := range []struct {
+		name     string
+		dest     map[string]interface{}
+		src      map[string]interface{}
+		expected map[string]interface{}
+	}{
+		{
+			name:     "Precedence",
+			dest:     m,
+			src:      override,
+			expected: override, // Expect the override to take precedence
+		},
+		{
+			name: "Merge maps",
+			dest: m,
+			src: map[string]interface{}{
+				"a": map[string]interface{}{
+					"b": map[string]interface{}{
+						"c": []interface{}{
+							"3", "4",
+						},
+						"f": true,
+					},
+				},
+			},
+			expected: map[string]interface{}{
+				"a": map[string]interface{}{
+					"b": map[string]interface{}{
+						"d": []interface{}{
+							"1", "2",
+						},
+						"c": []interface{}{
+							"3", "4",
+						},
+						"f": true,
+					},
+				},
+			},
+		},
+		{
+			name: "Dest Has Nil Values",
+			dest: m,
+			src: map[string]interface{}{
+				"a": map[string]interface{}{
+					"b": map[string]interface{}{
+						"c": interface{}(nil),
```

Comment:
Without the change, running the resulting merged map through `resource.NewPropertyMap()` would result in a assertion failure.